docs/typescript.md

```diff
@@ -4,18 +4,60 @@ Our use of TypeScript has to accomodate both .js development in agoric-sdk (whic
 
 ## Best practices
 
+- `.d.ts` for types modules
 - package entrypoint(s) exports explicit types
 - for packages upon which other packages expect ambient types:
   - `exported.js` exports the explicit types and ambient re-exports
 
+## .d.ts modules
+
+We cannot use `.ts` files any modules that are transitively imported into an Endo bundle. The reason is that the Endo bundler doesn't understand `.ts` syntax and we don't want it to until we have sufficient auditability of the transformation. Moreover we've tried to avoid a build step in order to import a module. (The one exception so far is `@agoric/cosmic-proto` because we codegen the types. Those modules are written in `.ts` syntax and build to `.js` by a build step that creates `dist`, which is the package export.)
```

Comment:
Is the `dist` not checked in? I didn't realize we introduced a required build step. At least those files don't really change much.

Reply:
Correct.